src/config/mod.rs

```diff
@@ -1376,6 +1377,44 @@ mod resource_tests {
         )
         .is_err());
     }
+
+    #[test]
+    #[ignore]
+    #[allow(clippy::print_stdout)]
+    #[allow(clippy::print_stderr)]
+    fn generate_component_config_schema() {
+        use crate::config::{SinkOuter, SourceOuter, TransformOuter};
+        use indexmap::IndexMap;
+        use vector_common::config::ComponentKey;
+        use vector_config::configurable_component;
+
+        /// Top-level Vector configuration.
+        #[configurable_component]
+        #[derive(Clone)]
+        struct ComponentsOnlyConfig {
+            /// Configured sources.
```

Comment:
Wouldn't these all be "configurable" or "available" (based on features) rather than configured?

Reply:
This is what the user has _specified_... as in, what they've chosen to configure in their topology.

Reply:
Gotcha, misunderstood - thought it was going to be "all components available in the binary"

Reply:
Yeah, no, for sure. It's also confusing how the big enums are called, like, `Transforms`, because there's just enough stuttering/duplication of name patterns that it'd hard to, with the name of types/fields alone, make it obvious that one thing is the list of all _possible_ transforms vs the the list of all transforms the user has _configured_, and so on.

It's fine for now, but your gut intuition that made you point this out in the first place.. it's not just you.